batch/batch/worker/worker.py

```diff
@@ -567,30 +765,68 @@ async def delete(self):
     #   timing: dict(str, float),
     #   error: str, (optional)
     #   short_error: str, (optional)
-    #   container_status: { (from docker container state)
+    #   container_status: {
     #     state: str,
     #     started_at: str, (date)
     #     finished_at: str, (date)
-    #     out_of_memory: bool
-    #     error: str, (one of error, exit_code will be present)
     #     exit_code: int
     #   }
     # }
     async def status(self, state=None):
         if not state:
             state = self.state
         status = {'name': self.name, 'state': state, 'timing': self.timings.to_dict()}
+        status['container_status'] = await self.get_container_status()
         if self.error:
             status['error'] = self.error
         if self.short_error:
             status['short_error'] = self.short_error
         if self.container_status:
             status['container_status'] = self.container_status
-        elif self.container:
+        elif self.container_is_running():
             status['container_status'] = await self.get_container_status()
```

Comment:
something seems wrong there, you call get_container_status twice

Reply:
Oops I think the top one should be removed